User_Auth/views.py

```diff
@@ -7,18 +7,30 @@
 from .serializer import LoginSerializer
 from .validator import verifying_user_login, verifying_signup_request
 
+
+
 @api_view(['POST'])
 def signup_api(request):
     if not verifying_signup_request(request):
         return Response({"Error": "Invalid request body"}, status=status.HTTP_400_BAD_REQUEST)
 
+    # Get the username from the request
+    username = request.data.get('username')
+
     # Create the user
-    user = User(username=request.data.get('username'))
+    user = User(username=username)
     user.set_password(request.data.get('password'))
-    user.save()
 
```

Comment:
write the condition if email is in the request then add that email else generate this(@yopmail.com) email